k9mail-library/src/test/java/com/fsck/k9/mail/transport/smtp/SmtpTransportTest.java

```diff
@@ -787,6 +793,38 @@ public void sendMessagePipelining_without354ReplyforData_shouldThrow() throws Ex
         server.verifyInteractionCompleted();
     }
 
+    @Test
+    public void sendMessagePipelining_with250and550ReplyforRecipients_shouldThrow() throws Exception {
+        Message message = getMessageWithTwoRecipients();
+        MockSmtpServer server = createServerAndSetupForPlainAuthentication("PIPELINING");
+        server.expect("MAIL FROM:<user@localhost>");
+        server.expect("RCPT TO:<user2@localhost>");
+        server.expect("RCPT TO:<user3@localhost>");
+        server.expect("DATA");
+        server.output("250 OK");
+        server.output("250 OK");
+        server.output("550 remote mail to <user3@localhost> not allowed");
+        server.output("354 End data with <CR><LF>.<CR><LF>");
+        server.expect(".");
+        server.output("554 no valid recipients given");
```

Comment:
I put it there based on the  examples in RFC . 
Anyway by success response you mean 250 ?